integration-libs/opf/base/root/services/opf-resource-loader.service.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DOCUMENT, isPlatformServer } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { ScriptLoader } from '@spartacus/core';
+
+import { throwError } from 'rxjs';
+import {
+  OpfDynamicScriptResource,
+  OpfDynamicScriptResourceType,
+} from '../model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfResourceLoaderService extends ScriptLoader {
+  constructor(
+    @Inject(DOCUMENT) protected document: any,
+    @Inject(PLATFORM_ID) protected platformId: Object
+  ) {
+    super(document, platformId);
+  }
+
+  protected readonly OPF_RESOURCE_ATTRIBUTE_KEY = 'data-opf-resource';
+
+  protected loadedResources: OpfDynamicScriptResource[] = [];
+
+  protected embedStyles(embedOptions: {
+    src: string;
+    callback?: EventListener;
+    errorCallback: EventListener;
+  }): void {
+    const { src, callback, errorCallback } = embedOptions;
+
+    const isSSR = isPlatformServer(this.platformId);
+
+    if (isSSR) {
+      return;
+    }
+
+    const link: HTMLLinkElement = this.document.createElement('link');
+    link.href = src;
+    link.rel = 'stylesheet';
+    link.type = 'text/css';
+    link.setAttribute(this.OPF_RESOURCE_ATTRIBUTE_KEY, 'true');
+
+    if (callback) {
+      link.addEventListener('load', callback);
+    }
+
+    if (errorCallback) {
+      link.addEventListener('error', errorCallback);
+    }
+
+    this.document.head.appendChild(link);
+  }
```

Comment:
[MEDIUM]

IMHO the `OpfResourceLoaderService` should not be responsible for implementing the low-level logic for loading styles. Instead, let's create a core service called `StylesLoader`, which will be placed next to the `ScriptsLoader`, and we can reuse it here. This approach will help adhere to the single responsibility principle.

Note: I remember we already discussed it and I said back then the styles-loading low level logic can stay in the OPF library (becasue it's not needed anywhere else _yet_). But I had second thoughts and concluded the styles-loading is a generic logic that should belong to the core/storefrontlib.